### DISCUSSION TITLE
adding a custom attr for a tdf file

### DISCUSSION THREAD

**Discussion by dbutler** [opentdf/platform#1204] [Platform Q&A] [answered]
I am attempting to add custom attributes to a tdf.

I was exploring the examples 
opentdf/platform/examples/cmd/encrypt.go and decrypt.go

        tdf, err := client.CreateTDF(tdfFile, strReader,
                //sdk.WithDataAttributes("https://example.com/attr/attr1/value/value1", "https://example.com/attr/attr2/value/value2"),
                sdk.WithKasInformation(
                        sdk.KASInfo{
                                // examples assume unsecure http
                                URL:       fmt.Sprintf("http://%s", cmd.Flag("platformEndpoint").Value.String()),
                                PublicKey: "",
                        }))

encrypt and decrypt work fine without adding attruibutes, but if I attempt to uncomment the sdk.WithDataAttributes line, I am able to encrypt
but not able to decrypt.  below is a snippet of the error I am seeing when I do the decrypt.  do I need to define the attr before it can be used?  where 
would that be done?  can you send link to full example and all steps needed?

time=2024-07-24T10:23:23.332-04:00 level=ERROR msg="Encountered database error" error=": err: no rows in result set (SQLSTATE 20000)"
time=2024-07-24T10:23:23.332-04:00 level=ERROR msg="could not get attribute by FQN" fqn=https://example.com/attr/attr1/value/value1 error="ErrNotFound: value not found\n: err: no rows in result set (SQLSTATE 20000)"
time=2024-07-24T10:23:23.332-04:00 level=ERROR msg="resource not found" error="ErrNotFound: value not found\n: err: no rows in result set (SQLSTATE 20000)" fqns="[https://example.com/attr/attr1/value/value1 https://example.com/attr/attr2/value/value2]"
time=2024-07-24T10:23:23.332-04:00 level=ERROR msg="rpc error: code = NotFound desc = resource not found" error="rpc error: code = NotFound desc = resource not found" fqns="https://example.com/attr/attr1/value/value1, https://example.com/attr/attr2/value/value2"
time=2024-07-24T10:23:23.332-04:00 level=ERROR msg="Error received from GetDecisions" err="rpc error: code = Internal desc = resource retrieval failed"
time=2024-07-24T10:23:23.332-04:00 level=WARN msg="Could not perform access decision!" err="authorization decision unexpected\nrpc error: code = Internal desc = resource retrieval failed"
Error: reader.doPayloadKeyUnwrap failed:  splitKey.rewrap failed:error making request to kas: error making rewrap request: rpc error: code = PermissionDenied desc = request error
rpc error: code = PermissionDenied desc = forbidden
Usage:
  examples decrypt [flags]

Flags:
  -h, --help   help for decrypt

Global Flags:
  -e, --platformEndpoint string   Platform Endpoint (default "localhost:8080")

reader.doPayloadKeyUnwrap failed:  splitKey.rewrap failed:error making request to kas: error making rewrap request: rpc error: code = PermissionDenied desc = request error
rpc error: code = PermissionDenied desc = forbidden

**Comment by jrschumacher** [comment, marked as answer]
@dbutler could you run `otdfctl --host $PLATFORM_ENDPOINT policy attributes list` and show me what attributes you've added to the system? This behavior is typical of setting up the system, but not having any attributes added. 

> [!NOTE]
> To manage your platform policy use otdfctl. To learn more run `otdfctl policy --help`, take note we use nested subcommands like `otdfctl policy attributes create --help`.
> For the CLI to reach out to the platform you need to specify a `--host` which is endpoint and port that your platform is reached at.
